app/src/main/java/com/activties/DetailsActivity.kt

```diff
@@ -38,34 +44,26 @@ class DetailsActivity : AppCompatActivity() {
 
     private fun loadMoviesCast(movieId: Int) {
         val apiKey = "ed0579c5972d5b789fec9a33235fcf3f"
-     RetrofitClient.apiService.getMoviesCredits(movieId,apiKey)
-         .enqueue(object : Callback<CreditsResponse> {
-                override fun onResponse(
-                    call: Call<CreditsResponse>,
-                    response: Response<CreditsResponse>
-                ) {
-                    if (response.isSuccessful) {
-                        response.body()?.cast?.let { credits->
-                            CastList.clear()
-                            CastList.addAll(credits)
-                            castAdapter = CastAdapter(CastList,this@DetailsActivity)
-                            castRecyclerView.adapter = castAdapter
-                        }
-                    }  else {
-                        Toast.makeText(this@DetailsActivity, "Error: ${response.code()}", Toast.LENGTH_SHORT).show()
-                    }
+        binding.progressBar.visibility = View.VISIBLE
+        CoroutineScope(IO).launch {
+            try {
+                val response = RetrofitClient.apiService.getMoviesCredits(movieId, apiKey)
+                withContext(Main) {
+                    CastList.clear()
+                    CastList.addAll(response.cast)
+                    castAdapter = CastAdapter(CastList, this@DetailsActivity)
+                    castRecyclerView.adapter = castAdapter
+                    binding.progressBar.visibility = View.GONE
                 }
-
-                override fun onFailure(call: Call<CreditsResponse>, t: Throwable) {
-                    Toast.makeText(this@DetailsActivity, "Failed to load Cast", Toast.LENGTH_SHORT).show()
+            } catch (e: Exception) {
+                withContext(Main) {
+                    binding.progressBar.visibility = View.GONE
+                    Toast.makeText(this@DetailsActivity, "Failed to load Cast: ${e.message}", Toast.LENGTH_SHORT).show()
```

Comment:
Good way of handling failures, please also give the coroutine exception handler a try to practice both of them.